cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -755,4 +780,105 @@ private void setWorkflowProperties() {
     properties.put(Constants.CONNECTOR_DATASETS, GSON.toJson(connectorDatasets));
     setProperties(properties);
   }
+
+  /**
+   *
+   * @param args the arguments to lookup spark engine custom config settings
+   * @param defaultProperties default value to use if spark engine custom config settings are missing from the arguments
+   */
+  public static Map<String, String> getProperties(Map<String, String> args,
+                                                  @Nullable Map<String, String> defaultProperties) {
```

Comment:
shouldn't allow a map be nullable, if it is null, the caller should pass in an empty map

Reply:
call this `originalProperties`, we don't have default value for these properties